components/form.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import Link from 'next/link';
+import { Person } from './Users';
+
+const baseUrl = 'https://rominacsvsms.api.ushahidi.io';
+
+export default class Form extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            token: {
+                access_token: '',
+                expires_in: 0,
+                refresh_token: '',
+                token_type: ''
+            },
+            formFields: [],
+            formData: {}
+        };
+    }
+
+    // Make state targets equal to the value of input fields
+    change = e => {
+        this.setState({
+            formData: {
+                ...this.state.formData,
+                [e.target.name]: e.target.value
+            }
+        });
+    };
+
+    isTokenValid() {
+        let isValid = false;
+        const currentTime = new Date().getTime();
+        const tokenFetchTime = this.state.token.fetching_time;
+        const expirationTime = this.state.token.expires_in;
+        if (currentTime < tokenFetchTime + expirationTime) {
+            isValid = true;
+        }
+        return isValid;
+    }
+
+    getToken() {
+        let token = JSON.parse(localStorage.getItem('Bearer Token'));
+        if (!token || this.isTokenValid()) {
+            fetch(baseUrl + '/oauth/token', {
+                method: 'POST',
+                body: JSON.stringify({
+                    scope: '*',
+                    client_secret: '35e7f0bca957836d05ca0492211b0ac707671261',
+                    client_id: 'ushahidiui',
+                    grant_type: 'password',
+                    password: Person.password,
+                    username: Person.userName
+                }),
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json',
+                    Authorization: 'large-auth-string'
+                }
+            })
+                .then(response => {
+                    if (response.ok) {
+                        return response.json();
+                    } else {
+                        throw new Error('Token fetching response is not ok.');
```

Comment:
What would be a more appropriate error message here?  what different types of responses can be expected and how would we represent them?

Reply:
changed it to include the status code with the error as well.